pkg/server/structlogging/hot_ranges_log_test.go

```diff
@@ -101,72 +89,50 @@ func TestHotRangesStats(t *testing.T) {
 			},
 		},
 	})
-	defer tc.Stopper().Stop(ctx)
-
-	db := tc.ServerConn(0)
-	sqlutils.CreateTable(
-		t, db, "foo",
-		"k INT PRIMARY KEY, v INT",
-		300,
-		sqlutils.ToRowFn(sqlutils.RowIdxFn, sqlutils.RowModuloFn(2)),
-	)
-
-	// Ensure both of node 1 and 2 have 5 distinct replicas from the table.
-	tableDesc := desctestutils.TestingGetPublicTableDescriptor(
-		tc.Server(0).DB(), keys.SystemSQLCodec, "test", "foo")
-	tc.SplitTable(t, tableDesc, []serverutils.SplitPoint{
-		{TargetNodeIdx: 1, Vals: []interface{}{100}},
-		{TargetNodeIdx: 1, Vals: []interface{}{120}},
-		{TargetNodeIdx: 1, Vals: []interface{}{140}},
-		{TargetNodeIdx: 1, Vals: []interface{}{160}},
-		{TargetNodeIdx: 1, Vals: []interface{}{180}},
-		{TargetNodeIdx: 2, Vals: []interface{}{200}},
-		{TargetNodeIdx: 2, Vals: []interface{}{220}},
-		{TargetNodeIdx: 2, Vals: []interface{}{240}},
-		{TargetNodeIdx: 2, Vals: []interface{}{260}},
-		{TargetNodeIdx: 2, Vals: []interface{}{280}},
-	})
 
-	// query table
-	for i := 0; i < 300; i++ {
-		db := tc.ServerConn(0)
-		sqlutils.MakeSQLRunner(db).Query(t, `SELECT * FROM test.foo`)
+	leakChecker := leaktest.AfterTest(t)
+	logInterceptor := log.InterceptWith(ctx, spy)
+	stopper := tc.Stopper()
+	teardown := func() {
+		stopper.Stop(ctx)
+		sc.Close(t)
+		logInterceptor()
+		leakChecker()
 	}
 
-	// Skip node 1 since it will contain many more replicas.
-	// We only need to check nodes 2 and 3 to see that the nodes are logging their local hot ranges.
-	rangeIDs := make(map[int64]struct{})
-	for _, i := range []int{1, 2} {
-		spy.Reset()
-		ts := tc.ApplicationLayer(i)
-		structlogging.TelemetryHotRangesStatsEnabled.Override(ctx, &ts.ClusterSettings().SV, true)
-		structlogging.TelemetryHotRangesStatsInterval.Override(ctx, &ts.ClusterSettings().SV, time.Second)
-		structlogging.TelemetryHotRangesStatsLoggingDelay.Override(ctx, &ts.ClusterSettings().SV, 0*time.Millisecond)
-
-		testutils.SucceedsSoon(t, func() error {
-			logs := spy.Logs()
-			if len(logs) < 5 {
-				return errors.New("waiting for hot ranges to be logged")
-			}
+	ts := tc.ApplicationLayer(0)
+	return ts, spy, teardown
+}
 
-			return nil
-		})
-		structlogging.TelemetryHotRangesStatsInterval.Override(ctx, &ts.ClusterSettings().SV, 1*time.Hour)
+// TestHotRangesStatsTenants tests that hot ranges stats are logged per node.
```

Comment:
nit: Commented test name doesn't match - also I'm not sure the comment actually reflects what's being tested anymore.

Reply:
I think the discrepency predates this PR. I'll change it though too.

Reply:
I'll change it.

Reply:
Whoops, wrong PR, this should be changed now.